impeller/entity/save_layer_utils.cc

```diff
@@ -106,8 +107,34 @@ std::optional<Rect> ComputeSaveLayerCoverage(
 
   // Transform the input coverage into the global coordinate space before
   // computing the bounds limit intersection.
-  return coverage.TransformBounds(effect_transform)
-      .Intersection(coverage_limit);
+  Rect transformed_coverage = coverage.TransformBounds(effect_transform);
+  std::optional<Rect> intersection =
+      transformed_coverage.Intersection(coverage_limit);
+  if (!intersection.has_value()) {
+    return std::nullopt;
+  }
+
+  // Sometimes a saveLayer is only slightly shifted outside of the cull rect,
+  // but is being animated in. This is common for the Android slide in page
+  // transitions. In these cases, computing a cull that is too tight can cause
+  // thrasing of the texture cache. Instead, we try to determine the
```

Comment:
```suggestion
  // thrashing of the texture cache. Instead, we try to determine the
```

Why does it cause thrashing?

Reply:
because the intersected size is different each frame, while the original size is stable

Reply:
Can you add that to the comment please?